### DISCUSSION TITLE
QST: GDAL 3.5+ supports GeoParquet/GeoArrow

### DISCUSSION THREAD

**Discussion by snowman2** [geopandas/geopandas#2724] [Q&A]
- https://gdal.org/drivers/vector/arrow.html
- https://gdal.org/drivers/vector/parquet.html

It would be interesting to see how the performance compares when reading using GDAL versus [read_parquet](https://geopandas.org/en/stable/docs/reference/api/geopandas.read_parquet.html) and [to_parquet](https://geopandas.org/en/stable/docs/reference/api/geopandas.GeoDataFrame.to_parquet.html). It is possible that those methods may not be necessary at some point in the future.

**Comment by jorisvandenbossche** [comment]
Our native `read_parquet`/`to_parquet` (using pyarrow directly, instead of through GDAL) are _much_ faster, so I don't think we will remove those anytime in the near future (and the same for feather).

It might be important to add something about this in the documentation, highlighting this that users should prefer the specialized methods instead of `read_file` for Parquet / Arrow files.

**Comment by snowman2** [comment]
That is interesting and good to know. I agree that it would be good to have that documented :+1:

**Comment by jorisvandenbossche** [comment]
To give one example to illustrate the order of magnitude difference (but the exact difference will depend a lot on the characteristics of the data):

```
In [2]: %time df = pyogrio.read_dataframe("gdal_geoparquet_bechmarks/nz-buildings-outlines.parquet")
CPU times: user 21.7 s, sys: 2.29 s, total: 24 s
Wall time: 23.9 s

In [3]: %time df = pyogrio.read_dataframe("gdal_geoparquet_bechmarks/nz-buildings-outlines.parquet", read_geometry=False)
CPU times: user 19.2 s, sys: 1.87 s, total: 21 s
Wall time: 21.2 s
```

vs 

```
In [1]: %time df = geopandas.read_parquet("gdal_geoparquet_bechmarks/nz-buildings-outlines.parquet")
CPU times: user 4.38 s, sys: 1.74 s, total: 6.11 s
Wall time: 4.61 s

In [3]: import pyarrow.parquet as pq

In [4]: %time table = pq.read_table("gdal_geoparquet_bechmarks/nz-buildings-outlines.parquet")
CPU times: user 1.6 s, sys: 698 ms, total: 2.3 s
Wall time: 814 ms

In [10]: columns_without_geo = list(df.columns.difference(["geometry"]))

In [11]: %time df2 = pd.read_parquet("gdal_geoparquet_bechmarks/nz-buildings-outlines.parquet", columns=columns_without_geo)
CPU times: user 1.37 s, sys: 237 ms, total: 1.6 s
Wall time: 583 ms
```

This is for a 440MB file (linked at the bottom of the README of https://github.com/opengeospatial/geoparquet/), and illustrating both reading all columns and reading all attribute columns without the geometry column. For (geo)pandas, I also compare it with reading it directly with pyarrow. That this is still much faster illustrates that the most time taken in `geopandas.read_parquet` is the conversion of the WKB values to GEOS objects.
